### PR TITLE
explicitly applying oom_score_adj(0) to processes of user containers

### PR DESCRIPTION
**Pull request by dchen1107**
related to #9788 

cc/ @yifan-gu 

With this PR, I tested it manually against a node, and all children processes of docker daemon are have either 0 or -100 as oom_score_adj.

```
# for i in $(docker ps -q); do out=$(docker top $i | grep -v UID); pid=$(echo $out | awk  -F" " '{print $2}'); cat /proc/$pid/oom_score_adj; done
0
0
0
0
0
0
0
0
0
-100
-100
-100
-100
-100
-100
```

I am working on a e2e test to prevent such issue from the future in a separate pr.

### CODE REVIEW

**Comment by dchen1107** [comment]
cc/ @brendandburns

**Comment by yifan-gu** [comment]
LGTM

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -1181,6 +1183,25 @@ func (dm *DockerManager) runContainerInPod(pod *api.Pod, container *api.Containe
...
+	// containers in the pod. This ensures that it is killed only as a last
+	// resort.
+	containerInfo, err := dm.client.InspectContainer(string(id))
+	if err != nil {
+		return "", err
+	}
+
+	// Ensure the PID actually exists, else we'll move ourselves.
+	if containerInfo.State.Pid == 0 {
+		return "", fmt.Errorf("failed to get init PID for Docker pod infra container %q", string(id))
```
Comment (yifan-gu) [inline pkg/kubelet/dockertools/manager.go]: failed to get the init PID for the Docker container

File: pkg/kubelet/dockertools/manager.go [hunk 2]
```diff
@@ -1181,6 +1183,25 @@ func (dm *DockerManager) runContainerInPod(pod *api.Pod, container *api.Containe
 	if err = dm.os.Symlink(containerLogFile, symlinkFile); err != nil {
 		glog.Errorf("Failed to create symbolic link to the log file of pod %q container %q: %v", podFullName, container.Name, err)
 	}
+
+	// Set OOM score of POD container to lower than those of the other
+	// containers in the pod. This ensures that it is killed only as a last
+	// resort.
```
Comment (yifan-gu) [inline pkg/kubelet/dockertools/manager.go]: Might want to update the comments here?

**Comment by yifan-gu** [comment]
Sorry, just found two small nit @dchen1107 , LGTM otherwise :)

File: pkg/kubelet/dockertools/manager.go [hunk 3]
```diff
@@ -1181,6 +1183,25 @@ func (dm *DockerManager) runContainerInPod(pod *api.Pod, container *api.Containe
...
+	}
+
+	// Ensure the PID actually exists, else we'll move ourselves.
+	if containerInfo.State.Pid == 0 {
+		return "", fmt.Errorf("failed to get init PID for Docker pod infra container %q", string(id))
+	}
+	if container.Name == PodInfraContainerName {
+		util.ApplyOomScoreAdj(containerInfo.State.Pid, podOomScoreAdj)
+	} else {
+		util.ApplyOomScoreAdj(containerInfo.State.Pid, userContainerOomScoreAdj)
```
Comment (yifan-gu) [inline pkg/kubelet/dockertools/manager.go]: Maybe add a comment here explaining why we need to explicit set the oom score adj here?

**Comment by yifan-gu** [comment]
Just small nits @dchen1107 , LGTM otherwise :)

**Comment by dchen1107** [comment]
Added comments, PTAL? Thanks!

**Comment by yifan-gu** [comment]
LGTM Thank you @dchen1107 !

**Comment by dchen1107** [comment]
ok to test

**Comment by dchen1107** [comment]
Chatted with @fabioy, merging it. @satnam6502 You can sync to head and build your 100 node cluster now.

**Pull request merged**